### DISCUSSION TITLE
DB2 DDL error...

### DISCUSSION THREAD

**Discussion by undertherainITA** [dbeaver/dbeaver#630] [Q&A] [answered]
Hi.
I have n.2 DB2 databases in the same instance, version 10.5, but I have two different behaviors when generating tables DDL.
One works perfectly, but the second returns the following error:

"**Errore SQL: [jcc][1037][12036][4.19.49] Exception occurred during clob conversion.  See attached Throwable for details. ERRORCODE=-4220, SQLSTATE=null
  [jcc][1037][12036][4.19.49] Exception occurred during clob conversion.  See attached Throwable for details. ERRORCODE=-4220, SQLSTATE=null
    java.nio.charset.MalformedInputException: Input length = 1**".

I use latest version of DB2 JDBC Driver 10.5 Fix 7.
Please help me...
Thank you so much for a great product... ;)

**Comment by titou10titou10** [comment]
Hi,
Could you send me the log files of DBeaver after reproducing the problem?
The files are in `"<user>\.dbeaver\.metadata"` named `"dbeaver-debug.log"` and `".log"`
Thanks
Denis (author of the DB2 plugin: titou10.titou10@gmail.com)

**Comment by undertherainITA** [comment]
Hi.
Sorry for the late response.
I run the DDL generation on 2 structurally identical databases that are both on unique 10.5 DB2 instance.
KSOSVIAL creates the DDL correctly, while KSOSVI2S returns this error:

"**Errore SQL: [jcc][1037][12036][4.19.49] Exception occurred during clob conversion.  See attached Throwable for details. ERRORCODE=-4220, SQLSTATE=null [jcc][1037][12036][4.19.49] Exception occurred during clob conversion.  See attached Throwable for details. ERRORCODE=-4220, SQLSTATE=null java.nio.charset.MalformedInputException: Input length = 1**".

Thanks!!!
Ciao... ;)
[dbeaver-debug.txt](https://github.com/serge-rider/dbeaver/files/393204/dbeaver-debug.txt)

**Comment by titou10titou10** [comment]
Hi,
Could you send me the DBeaver/eclipse log file that should contains the java stacktrace when the problem occurred. It's name is `"<user>\.dbeaver\.metadata\.log"`
You can send it directly to me on my email address: titou10.titou10@gmail.com
Denis

**Comment by undertherainITA** [comment]
Sent... ;)

**Comment by titou10titou10** [comment, marked as answer]
After investigation (and your help!), it appears that the problem is caused by some UTF-8 characters held in the database that the DB2 JDBC driver can not decode, and not from DBeaver itself. The problem will arise with any java program that will access those charcaters via the jdbc driver. 

This is explained in those IBM technotes:
- http://www-01.ibm.com/support/docview.wss?uid=swg21684365
- http://www-01.ibm.com/support/docview.wss?uid=swg1IC74896

To solve the problem: 
- edit the `dbeaver.ini` file (in the directory where you installed DBeaver)
- at the very end of the file add the 2 following lines (exactly as written):

`-vmargs`
`-Ddb2.jcc.charsetDecoderEncoder=3`
